### DISCUSSION TITLE
Error message......PrismaClient is unable to be run in the browser

### DISCUSSION THREAD

**Discussion by Rutancho** [prisma/prisma#24701] [Q&A]
### Question

I got this kinds of error message how can I solve this problem?

![asdsadd](https://github.com/prisma/prisma/assets/128659866/816932bc-5fdf-4854-a99c-0e2fb403a2fb)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Rutancho 

The error `PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in unknown)` typically occurs when you try to run Prisma Client in a browser environment or if it has been bundled for the browser. Prisma Client is designed to run in a server-side environment and does not work on the client side as the browser does not execute SQL queries.

If you are using Next.js, there are a couple of solutions you can try:

1. If you are using the Next.js app router, add "use server" at the top of your file.
2. If you are using the Next.js pages router, keep the files that import Prisma or contain any Prisma query function inside the /api directory.
3. If you are trying to use Prisma in a client component, you might encounter this error. Ensure that Prisma is used in a server component

You can also take a look [here](https://github.com/prisma/prisma/issues/6219) to see how others resolved this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
